dotnet/src/Connectors/Connectors.Amazon/Bedrock/Core/IBedrockModelIOService.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Text.Json.Nodes;
+using Amazon.BedrockRuntime.Model;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace Microsoft.SemanticKernel.Connectors.Amazon.Core;
+
+/// <summary>
+/// Bedrock input-output service to build the request and response bodies as required by the given model.
+/// </summary>
+internal interface IBedrockModelIOService
```

Comment:
Since this internal interface is enforcing you to implement `NotSupported`, suggest splitting this interface per modality.

`IBedrockChatCompletionIOService`
`IBedrockTextGenerationIOService`